web/src/util/line-highlight/prism-line-highlight.ts

```diff
@@ -0,0 +1,136 @@
+import Prism from "prismjs";
+
+// This file is a modified copy of primsjs's prism-line-highlight plugins which exposes utility functions
```

Comment:
Why can't we access the utility functions directly? What did you modify?

Reply:
All the functions used in the plugin are created inside another function, which is called as soon as you import the plugin. Thus not exposing any of the functions used in the plugin itself. The things that I have modified so far, apart from adding typing and fixing linting errors, is mainly adding the onClick event handler that can be given via the Highlighting options.